src/compiler/factory/nodeFactory.ts

```diff
@@ -7570,6 +7583,9 @@ export function createSourceMapSource(fileName: string, text: string, skipTrivia
 
 // Utilities
 
+/** @internal */ export function setOriginalNode<T extends Node>(node: Mutable<T>, original: Node | undefined): T; // TODO: This should probably always require a Mutable<T>, since it mutates `original`
+// eslint-disable-next-line @typescript-eslint/unified-signatures

```

Comment:
Is there a bugfix or feature request you'd want to file on typescript-eslint to make `unified-signatures` not complain on this case?